Original patch by Dominique Martinet <dominique.martinet@atmark-techno.com>

BPF_MAP_TYPE_TASK_STORAGE and BPF_PROG_TYPE_SYSCALL were added in kernels newer
than 5.10 and are not available to us yet.

diff --git a/introspection/bps.c b/introspection/bps.c
index 0eae675e3817..b5595442faee 100644
--- a/introspection/bps.c
+++ b/introspection/bps.c
@@ -79,7 +79,6 @@ static const char * const map_type_strings[] = {
   [BPF_MAP_TYPE_STRUCT_OPS] = "struct_ops",
   [BPF_MAP_TYPE_RINGBUF] = "ringbuf",
   [BPF_MAP_TYPE_INODE_STORAGE] = "inode_storage",
-  [BPF_MAP_TYPE_TASK_STORAGE] = "task_storage",
 };
 
 #define ARRAY_SIZE(x) (sizeof(x) / sizeof(*(x)))

diff --git a/introspection/bps.c b/introspection/bps.c
index 6ec02e6c..0eae675e 100644
--- a/introspection/bps.c
+++ b/introspection/bps.c
@@ -47,7 +47,6 @@ static const char * const prog_type_strings[] = {
   [BPF_PROG_TYPE_EXT] = "ext",
   [BPF_PROG_TYPE_LSM] = "lsm",
   [BPF_PROG_TYPE_SK_LOOKUP] = "sk_lookup",
-  [BPF_PROG_TYPE_SYSCALL] = "syscall",
 };

 static const char * const map_type_strings[] = {
